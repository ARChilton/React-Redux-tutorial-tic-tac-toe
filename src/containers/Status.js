import { connect } from 'react-redux'

export const Status = ({status = 'Next player is O'}) => (
  <div>
    {status}
  </div>
)

//props req. status and moves
export const mapStateToProps = (state) => ({
  status: state.xoGame.winner
    ? 'Winner: ' + state.xoGame.winner
    : 'Next player: ' + (state.xoGame.xIsNext ? 'X' : 'O'),
})

export default connect(
  mapStateToProps,
)(Status)
