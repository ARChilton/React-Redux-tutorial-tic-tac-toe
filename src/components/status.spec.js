import React from 'react'
import renderer from 'react-test-renderer'
import Status from './status'

it('renders correctly', () => {
  const statusText = 'test status'
  const status = renderer
    .create(<Status status={statusText} />)
    .toJSON()
  expect(status).toMatchSnapshot()
})
