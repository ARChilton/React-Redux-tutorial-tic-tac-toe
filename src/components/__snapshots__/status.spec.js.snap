// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    test status
  </div>
</div>
`;
